core/pkg/ingress/controller/controller.go

```diff
@@ -0,0 +1,1076 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
can you move this into ingress/controller/generic/
with 
README
interfaces.go
controller.go
tests...
etc?